.github/workflows/reusable_go_lint_test.yml

```diff
@@ -0,0 +1,70 @@
+name: Reusable Go Build, Lint, and Test Workflow
+
+on:
+  workflow_call:
+    inputs:
+      go-version:
+        description: 'Go version'
+        type: string
+        default: '1.22.3'
+      install-dependencies-command:
+        description: 'Command to install dependencies'
+        required: false
+        type: string
+      run-unit-tests:
+        description: 'Run unit tests'
+        type: boolean
+        default: false
+      run-integration-tests:
+        description: 'Run integration tests'
+        type: boolean
+        default: false
+
+jobs:
+  build_lint_test:
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: ${{ inputs.go-version }}
+
+      - name: Print Go environment
+        run: go env
+
+      - name: Cache Go modules
+        id: go-cache
+        uses: actions/cache@v4
+        with:
+          path: |
+            ~/.go/pkg/mod
+            ~/.cache/go-build
+          key: go-mod-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            go-mod-
+
+      - name: Install Dependencies
+        if: ${{ inputs.install-dependencies-command != '' }}
+        run: ${{ inputs.install-dependencies-command }}
+
+      - name: Build Application
+        run: make build
+
+      - name: Lint
+        run: |
+          curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s v1.55.2
```

Comment:
Does the go linter version depend on the go version?
If so (probably), let's also templatize this version.